includes/wikia/Wikia.php

```diff
@@ -2448,6 +2451,32 @@ public static function getCountryNames( array $countryCodes ) {
 	}
 
 	/**
+	 * Hide certain namespaces from list of movable ones
+	 *
+	 * @author pgroland
+	 * @see CE-1378
+	 *
+	 * @param index Index of namespace
+	 * @return bool
+	 */
+	static public function isNamespaceMovable( &$result, $index ) {
+		// namespace ids from extensions are hardcoded here
```

Comment:
Is there any reason of having the namespaces hardcoded here?

The only reason I see, is that the named constants can only be used when corresponding extensions are enabled (and the code defining those constants is included).

I would refactor the entire thing so that the extensions in question automatically add their namespaces to the list of movable namespaces. But that touches on the topic of the global state. Another option is to have extensions' own functions hooked to `MWNamespace:isMovable' event covering only the relevant extensions' namespaces. It should be simple as the only thing we do here is to modify `$result`. Depending on the context, the execution of the hook could be interrupted once non-movable namespace is detected.